text/0000-permissions.md

```diff
@@ -0,0 +1,186 @@
+- Feature Name: `permissions`
+- Start Date: 2023-01-30
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+
+# Summary
+[summary]: #summary
+
+Set permissions for functions and add permissioned borrows.
+
+# Motivation
+[motivation]: #motivation
+
+So I've been latetly writing a lot of rust code where I need a struct mutable borrowed and stored paralelly to other borrows of the same data.
+They really don't interop or block each other, they are two iterators nested and one needs the whole type to call a function on it (immutable) and get that data.
+The other iterator needs a mutable borrow of an underlying `HashMap` in that same struct to convert the key to a mutable ref to the value.
+But these two iterators nested wouldn't work, because they would both reference to the same data.
+To resolve the error: ```cannot borrow `*app` as mutable more than once at a time
+second mutable borrow occurs here```, I've just be using unsafe pointer magic..:
+```rust
+// ..with a safety comment above
+let ptr: *mut G = &mut *app;
+let inner = Self::new(&*ptr);
+```
+While I wrote this, the idea of a langauge with permissions came up to me.
+
+> The explanation of how it should work follows in the next chapter.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+So, a permission is looking like a life-time specifier.
```

Comment:
This feels out of place. You start by saying that a permission looks like a lifetime, but then you don't continue to describe permissions, but rather first define some example types/functions.